---
title: 'Brief Introduction to Genomic informed Drug Target Database'
author: "BALDER Team^[Merina Shrestha, Zhonghao Bai, Tahereh Gholipour, Viktor Milkevych, Peter Sørensen, Center for Quantitative Genetics and Genomics, Aarhus University, Denmark. Palle Duun Rohde, Genomic Medicine, Department of Health Science and Technology, Aalborg University, Denmark. Astrid Johanneson Hjelholtz, Mads Fuglsang Kjølby, Department of Biomedicine, Aarhus Univertsity, Denmark. Project funding: ODIN (NNF20SA0061466).]"
date: "`r Sys.Date()`"
bibliography: qg2021.bib
biblio-style: apalike
link-citations: yes
output:
  pdf_document:
    dev: png
    includes:
      in_header: preamble.tex
  html_document:
    includes:
      in_header: mathjax_header.html
---

```{r, echo = F}
#install.packages("formatR", repos = "http://cran.rstudio.com")
library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE)
eval <- TRUE
echo_solution <- TRUE
message <- FALSE
```

# Introduction {.unlisted .unnumbered} 
Here we release scripts and workflows that provide an infrastructure for efficiently working with large-scale genomic association data, specifically linked to various types of functional marker sets obtained from publicly available resources. This framework enables the exploration of novel genetic markers associated with complex diseases such as Type 2 Diabetes, facilitating the identification of biomarkers and potential drug targets.

One key aspect is the investigation of genes, regulatory elements, and biological pathways that are closely associated with disease-associated variants. Additionally, we explore biomarkers (including transcriptome, metabolome, and proteome) whose association signals are linked to disease-related signals. This comprehensive approach aids in the discovery of biomarkers and potential drug targets, ensuring an informed drug target discovery process. By studying the impact of genetic variants directly in humans, we can better assess the effectiveness of drugs targeting specific biological processes, surpassing reliance solely on computer or animal models.

To accomplish this, we download and process functional marker information from various sources, including Ensembl (linking SNPs to genes and proteins), GO (Gene Ontology), STRING (protein-protein interactions), STITCH (protein-chemical interactions), Reactome (biological pathways), and the GWAS Catalog. Our processing pipeline includes quality control of GWAS summary statistics, mapping to LD reference panels, and the creation of marker sets linked to genes, proteins, and pathways. These marker sets facilitate marker set analyses and aid in the biological interpretation of genome-wide association studies, specifically by assessing the association of functional marker sets (e.g., biological pathways, protein complexes, gene ontology terms) with complex diseases.

Moreover, our workflow allows for the testing of specific biological hypotheses. For example, we can assess if genes, proteins, metabolites, and pathways underlying Type 2 Diabetes are enriched for association signals with the disease. Additionally, we can explore if drugs used in the treatment of Type 2 Diabetes are associated with genes, proteins, metabolites, and pathways that display enriched association signals.

By continually incorporating new functional marker sets into our workflow, we ensure the ability to identify and process relevant data for Type 2 Diabetes and other complex diseases, providing valuable insights for ongoing research.


## Load packages used {.unlisted .unnumbered} 
The practical is based on the R package `gact` (Sørensen et al.2022). 
The most recent version of `gact` can be obtained from github:

```{r,  eval=FALSE, echo=TRUE}
library(devtools)
devtools::install_github("psoerensen/gact")
```

```{r,  eval=eval, echo=TRUE}
library(gact)
library(qgg)
library(corrplot)
library(data.table)
```

## Download and install GDT database {.unlisted .unnumbered} 
The function `gact()` dowload and install the GDT database: 
```{r,  eval=FALSE, echo=TRUE}
# Set working for database
dbdir <- "C:/Projects/balder/gdtdb"

# Download the database from repository
GAlist <- gact(version="gact-0.0.1", dbdir=dbdir, task="download")
```

```{r,  eval=TRUE, echo=FALSE}
dbdir <- "C:/Users/au223366/Dropbox/Projects/balder/gdtdb"
GAlist <- readRDS(file=file.path(dbdir, "GAlist_gact-0.0.1.rds"))
```


## Study and feature information in the database {.unlisted .unnumbered} 
```{r,  eval=TRUE, echo=TRUE}
# Information about studies in GDT database
head(as.data.frame(GAlist$study))

# Information about features in GDT database
GAlist$features
```


## Extract data from GDT database {.unlisted .unnumbered} 
The function `getStatDB()` extract data from the database: 
```{r,  eval=TRUE, echo=TRUE}

# Extract marker summary statistics GWAS1
stat <- getMarkerStatDB(GAlist = GAlist, studyID = "GWAS1")
head(stat)
rownames(stat) <- stat$rsids

# Extract marker sets for gene ENSG00000147883
sets <- getSetsDB(GAlist = GAlist, feature = "Genes", featureID = "ENSG00000147883")
# sets is a list
str(sets)
# convert sets into a vector
rsids <- sets[[1]]

# Plot -log10 p-values for selected gene
plot(y = -log10(stat[stat$rsids %in% rsids, "p"]), x = stat[stat$rsids %in%
    rsids, "pos"], ylab = "-log10(p)", xlab = "Position", frame.plot = FALSE,
    main = "ENSG00000147883")

# Extract data for GWAS1 for genomic feature Genes
stat <- getStatDB(GAlist = GAlist, studyID = "GWAS1", feature = "Genes",
    format = "data.frame")
head(stat)

# Plot results
plot(x = stat$stat/stat$m, y = -log10(stat$p), ylab = "-log10 p-values",
    xlab = "Average signal per SNP", frame.plot = FALSE, main = "Genes")

# Extract data for GWAS1 for genomic feature Gene Ontology (GO) where
# output format is a data frame
stat <- getStatDB(GAlist = GAlist, studyID = "GWAS1", feature = "GO")
str(stat)
plot(x = stat$stat/stat$m, y = -log10(stat$p), ylab = "-log10 p-values",
    xlab = "Average signal per SNP", frame.plot = FALSE, main = "GO")

# Extract data from T2D for genomic feature Gene Ontology (GO) where
# output format is list
stat <- getStatDB(GAlist = GAlist, feature = "GO")
str(stat)

library(corrplot)

# Plot results for genomic feature Gene Ontology (GO)
colbar <- colorRampPalette(c("#FFFFFF", "#D1E5F0", "#92C5DE", "#4393C3",
    "#2166AC", "#053061"))
corrplot(-log10(stat$p[1:10, ]), is.corr = FALSE, tl.cex = 0.7, tl.srt = 45,
    col = colbar(6), cl.pos = "n", mar = c(1, 1, 1, 1))

# Extract results from gsea of Reactome pathways
stat <- getStatDB(GAlist = GAlist, feature = "Genes", studyID = "GWAS1")
str(stat)

# Select the top pathways for GWAS1
featureID <- names(stat$p[stat$p < 0.01])
stat <- getStatDB(GAlist = GAlist, feature = "Genes", studyID = "GWAS1",
    featureID = featureID)
str(stat)

# Get marker stat for GWAS1 and GWAS2
stat <- getMarkerStatDB(GAlist = GAlist, studyID = c("GWAS1", "GWAS2"))
str(stat)

# Get gene-level association statistics (i.e. sum of )
stat <- getStatDB(GAlist = GAlist, feature = "Genes", threshold = 0.95,
    studyID = "GWAS1")
y <- scale(stat$stat)

# Adjust for gene length
y <- residuals(lm(y ~ stat$m))
hist(y)

# Get design matrix for gene sets
W <- designMatrixDB(GAlist = GAlist, feature = "Pathways", rowFeatureID = names(y))
W <- scale(W)


# Fit BLR model
fit <- gbayes(y = y, W = W, method = "bayesC")

# Plot some BLR result
layout(matrix(1:2, ncol = 2))
plot(fit$dm)
plot(fit$bm)
```

## Extract and write data from GDT database {.unlisted .unnumbered} 
The function `writeStat()` extract and write data from the database: 
```{r,  eval=FALSE, echo=TRUE}
writeStatDB(GAlist=GAlist, feature="GO", studyID="GWAS1", file.csv="go_GWAS1_gcta.csv")
writeStatDB(GAlist=GAlist, feature="Pathways", studyID="GWAS1", file.csv="pathways_GWAS1_gcta.csv")
writeStatDB(GAlist=GAlist, feature="ProteinComplexes", studyID="GWAS1", file.csv="proteincomplexes_GWAS1_gcta.csv")
writeStatDB(GAlist=GAlist, feature="ChemicalComplexes", studyID="GWAS1", file.csv="chemicalcomplexes_GWAS1_gcta.csv")
writeStatDB(GAlist=GAlist, feature="Genes", studyID="GWAS1", file.csv="genes_GWAS1_gcta.csv")
```

## Extract marker set data from GDT database {.unlisted .unnumbered} 
The marker sets in the database can be extracted using: 
```{r,  eval=TRUE, echo=TRUE}
geneSets <- getSetsDB(GAlist=GAlist,feature="Genes")
chemSets <- getSetsDB(GAlist=GAlist,feature="ChemicalComplexes2Genes")

```

## Extract data for chemical "CIDm00004091" from GDT database {.unlisted .unnumbered} 
The marker sets in the database can be extracted using: 
```{r,  eval=TRUE, echo=TRUE}
chemStat <- getStatDB(GAlist=GAlist, studyID="GWAS1", feature="ChemicalComplexes", format="data.frame")
chemStat["CIDm00004091",]

genesStat <- getStatDB(GAlist=GAlist, studyID="GWAS1", feature="Genes", format="data.frame")
sets <- getSetsDB(GAlist=GAlist, feature="ChemicalComplexes2Genes", featureID="CIDm00004091")
# sets is a list
str(sets)
ensgIDs <- sets[[1]]
# convert sets into a vector
ensgIDs <- sets[[1]]
#select rsids in stat object
ensgIDs <- ensgIDs[ensgIDs%in%rownames(genesStat)]

head(genesStat[ensgIDs,])
```


## Process external GWAS summary statistic data and add to database {.unlisted .unnumbered} 
library(data.table)

```{r,  eval=FALSE, echo=TRUE}
# Load GAlist
GAlist <- readRDS(file = file.path(dbdir, "GAlist_gact-0.0.1.rds"))

# Add Meta-analysis of BMI in UK Biobank and GIANT data.  Combined
# set of samples, max N = 697,734.
fname_stat <- "https://portals.broadinstitute.org/collaboration/giant/images/1/14/Bmi.giant-ukbb.meta-analysis.combined.23May2018.HapMap2_only.txt.gz"
stat <- fread(fname_stat, data.table = FALSE)
stat <- stat[, c("SNP", "CHR", "POS", "Tested_Allele", "Other_Allele",
    "Freq_Tested_Allele", "BETA", "SE", "P")]
colnames(stat) <- c("marker", "chr", "pos", "ea", "nea", "eaf", "b", "seb",
    "p")
head(stat)

GAlist <- updateStatDB(GAlist = GAlist, 
                       stat = stat, 
                       source = "bmi.giant-ukbb.meta-analysis.combined.23May2018.txt.gz",
                       trait = "BMI", 
                       type = "quantitative", 
                       gender = "both", 
                       ancestry = "White",
                       build = "GRCh37", 
                       reference = "PMID:30239722", 
                       n = 505454, 
                       ncase = 0,
                       ncontrol = 0, 
                       comments = "Meta-analysis", 
                       writeStatDB = TRUE, 
                       excludeMAFDIFF = 0.05)

# Save updated GAlist
saveRDS(GAlist, file = file.path(dbdir, "GAlist_gact-0.0.1.rds"))
```


## Shiny Apps to use genomic informed drug target database {.unlisted .unnumbered} 
These app provides an interface for conducting a range of
gene set enrichment analyses and visualizing the results in
various formats.The interface includes a title panel and a sidebar
layout with input controls for selecting the study, entering gene
symbols, and specifying analysis parameters. The main panel
displays the results of the analysis in several tabs, including
data tables, a Manhattan plot, and information about
available studies.

```{r,  eval=FALSE, echo=TRUE}
# Load GAlist
GAlist <- readRDS(file=file.path(dbdir, "GAlist_gact-0.0.1.rds"))

# Shiny App for Gene Set Enrichment Analysis of Gene Complexes
shinyAppsDB(GAlist=GAlist, what="customGSEA")

# Shiny App for Gene Set Enrichment Analysis of Genomic Features
shinyAppsDB(GAlist=GAlist, what="featureGSEA")

# Shiny App for Drug Target Analysis of a Specific Genomic Features
shinyAppsDB(GAlist=GAlist, what="featureDRUGS")
```




